### ISSUE TITLE
Android 7.0: crash at startup (before screen opens)

### ISSUE THREAD

**Issue by realnessy** [Freeyourgadget/Gadgetbridge#1106]
#### Before opening an issue please confirm the following:
- [x ] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x ] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x ] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:
As soon as I start the app, it crashes, and my smartphone displays "Gadgetbridge wurde beendet / App wieder öffnen" ("Gadgetbridge has been terminated / restart app).
I don't know how to get logs of this behaviour...
I just updated to 0.26.5, as it also happened with the previous version. I hoped that maybe this has been fixed, but it is still there...
I have the same app running on a different smartphone with Android 4.4, and there it works (activities don't work, I will submit a different bug report)
I should mention that I don't use a google account on my Android 7.0, so in case you expect a google account, this might be the problem.
Other than that, I have no idea what could be wrong, my smartphone runs really fine, no other app shows this behaviour.
Smartphone is a "Outikel K10000 Max"

#### Your wearable device is:

Amazfit Bip (but there was no chance to pair it, so I don't think it is relevant)

#### Your android version is:
7.0

#### Your Gadgetbridge version is:
0.26.5

*New issues about already solved/documented topics could be closed without further comments. Same for too generic or incomplete reports.*

**Comment by danielegobbetti** [comment]
Can you try going into the system settings -> apps -> gadgetbridge and check the granted permissions? Then grant them all from there.

I'm wondering if there is something in the ROM of your phone that prevents us from getting the permissions dialogs to show, but it's a wild guess.

**Comment by realnessy** [comment]
@danielegobbetti 
About permissions: when there are no permissions set, then the crash happens WHILE the permission windows are shown. I see the crash window, and underneath there are the permissions. It asks for FIVE permissions, and when I go to settings, then only GPS is *not* permitted, so obviously it never reached the point where it asks for the GPS permission.
I granted all permissions (including GPS), but it still crashes (then showing now permission windows, as the permissions have already been granted).
There are two more things that I might have to share:
1) This is a multiuser smartphone, I can change between users, like on tablets. I tried with the main user, but there are two more users configured. No idea if this makes some difference, internally...
2) That smartphone often runs in a different time zone (not my home timezone). It runs in GMT+13 (for some online game...). I know it will change my clock, and maybe I turn the time zone back. to CEST once I get this to run.
3) Are you using any internet connection at that time of the startup? I have Pi-Hole running, and if any advertise-server is trying to be connected, there's a chance that my Pi-Hole redirects the request to localhost, if it thinks that the request is for an ad server. Normally, this does not cause problems (just less ads to be shown), but I better mention it. I can make an exception for an ad server, if necessary. Pi-Hole basically is a DNS server, running in my LAN, and resolving ad-servers to 127.0.0.1
Other than that, there's really not much installed on my phone, it is quite new

**Comment by danielegobbetti** [comment]
Gadgetbridge doesn't connect to the internet (the internet permission is not even requested). Can you try starting the app with Bluetooth turned off to see if it makes any change?

**Comment by realnessy** [comment]
@danielegobbetti
I turned off Bluetooth in the settings as requested, but the app still crashes.

**Comment by danielegobbetti** [comment]
I see in your log that you have no webview installed. I don't think I ever saw an android device without a webview, are there some custom protections in your phone's operating system?

Can you try going to the system settings -> developers tools -> webview implementation(or webview provider) and see what is selected and available in that menu? Also please try starting gadgetbridge again after making a selection.

**Comment by realnessy** [comment]
I do find the option, but clicking it won't allow me to select anything. It is just a window with a title and a "Cancel" option.

I am not familiar with Android programming. What is WebView? Some HTML viewer?
I have e.g. Telegram running, and when clicking on a link, it opens some Web View window (but not the Firefox or Chrome that I have installed). Both, Firefox and Chrome are working fine.

Is there a way how I can install that "WebView" thing, so that maybe I have something to select in the developer options?

**Comment by danielegobbetti** [comment]
Yes a webview is a browser engine that is provided by the system and might be embedded by other apps. As far as I know it  is available on google play store (android system webview) but there are some restrictions on its availability (it's a bit technical, but its signature must be compatible with the one of your installed OS). Unfortunately it is not available on f-droid for this very reason, so if you don't have the play store available I don't know how to help you.

We can try to make it optional, since it's only needed for some gadgets, but that will take some time.

**Comment by realnessy** [comment]
Strange, in the settings, if I list system apps, then there is a "Android System WebView" showing up. It is version 0.0.0.1 (?)
I got the arm version from apkmirror, and I also updated google play services (just because I know there is a similar issue with updates). Now the crash is gone - yeah, and WebView is listed with a reasonable version number in settings.
Thank you very much for helping to solve this!
For other people with the same problem it will probably be helpful to  make WebView non-mandatory, or to at least show some error message. ;-)
